classes/class-wc-connect-help-provider.php

```diff
@@ -145,6 +145,14 @@ protected function get_health_items() {
 					__( 'Please connect Jetpack to WordPress.com', 'woocommerce' ),
 					''
 				);
+			} else if ( Jetpack::is_staging_site() ) {
```

Comment:
If Jetpack is not connected the ```else if``` branches above this one will execute instead, so this won't ever get checked if Jetpack is not connected / installed.